server.js

```diff
@@ -0,0 +1,133 @@
+// Load Environment Variables from the .env file
+require('dotenv').config();
+
+// Application Dependencies
+const express = require('express');
+const cors = require('cors');
+const morgan = require('morgan');
+const client = require('./lib/client');
+// Initiate database connection
+client.connect();
+
+//Auth
+const ensureAuth = require('./lib/auth/ensure-auth.js');
+const createAuthRoutes = require('./lib/auth/create-auth-routes.js');
+const authRoutes = createAuthRoutes({
+    selectUser(email){
+        return client.query(`
+            SELECT id, email, hash, shown_name as shownName
+            FROM users
+            WHERE email = $1
+        `, [email])
+            .then(result => result.rows[0]);
+    },
+    insertUser(user, hash){
+        return client.query(`
+            INSERT INTO users (email, hash, shown_name)
+            VALUES ($1, $2, $3)
+            RETURNING id, email, shown_name as "shownName";
+        `, [user.email, hash, user.shownName])
+            .then(result => result.rows[0]);
+    }
+});
+
+// Application Setup
+const app = express();
+const PORT = process.env.PORT;
+app.use(morgan('dev')); // http logging
+app.use(cors()); // enable CORS request
+app.use(express.static('public')); // server files from /public folder
+app.use(express.json()); // enable reading incoming json data
+
+app.use('/api/auth', authRoutes);
+app.use('/api', ensureAuth);
+// API Routes
+
+// *** TODOS ***
+app.get('/api/todos', async (req, res) => {
+
+    try {
```

Comment:
this needs to be updated as well to only get todos for the currently logged in user

`try {
        const result = await client.query(`
            SELECT * FROM todos
            WHERE user_id=$1
        `,
        [req.userId]
        );

        res.json(result.rows);
    }`